docs/DevGuide/CreatingExecutables.md

```diff
@@ -16,7 +16,109 @@ There are several different types of executables that can be built:
+SingletonHelloWorld.hpp which is found in the directory
+`src/Executables/Examples/HelloWorld`.  `Metavars` will be used as the
+metavariables struct, and `SingletonHelloWorld` will link against the `%Informer`
+and `Utilities` SpECTRE libraries.
+
+The header file from which an executable is generated must define a metavariables
+struct that can be thought of as a compile-time input file that defines what the
+executable will do.  The `Metavars` struct for `SingletonHelloWorld` is a
+minimal example of a metavariables struct.
+
+\snippet SingletonHelloWorld.hpp executable_example_metavariables
+
+Each metavariables must define an enum class `Phase` with the phases of the
+executable (which must include `Initialization` and `Exit`), and a static
+function `determine_next_phase`.  In the example, the only additional phase is
+`Execute`, and the phases are executed in order.  Charm++ will execute each phase
+until it detects that nothing is happening (quiescence detection).  As this
+represents a gloabl synchronization point, the number of phases should be
+minimized in order to exploit the power of SpECTRE.
+
+Each metavariables must define a type alias `component_list` that is a
+`tmpl::list` of the parallel components used by the executable.
+`SingletonHelloWorld` defines a single component `HelloWorld`
+
+\snippet SingletonHelloWorld.hpp executable_example_singleton
+
+which specifies via the `chare_type` type alias that it is a singleton parallel
+component which means that only one such object will exist across all processors
+used by the executable.  Each component must define the static functions
+`initialize`, which is executed during the `Initialization` phase, and
+`execute_next_phase` which is executed during the phases (other than
+`Initialization` and `Exit`) defined in the metavariables struct).  In
+`SingletonHelloWorld`, nothing is done during the initialization phase, while the
+`PrintMessage` action is called during the `Execute` phase.
+
+\snippet  SingletonHelloWorld.hpp executable_example_action
+
+The `PrintMessage` action is executed on whatever process the singleton component
+is created upon, and prints a message.
+
+Executables can read in an input file (specified by the `--input-file` argument)
+that will be parsed when the executable begins.  %Options specified in the input
+file can be used to either place items in a constant global cache (by specifying
+options in the `const_global_cache_tag_list` of the metavariables or component
+structs) or be passed to the `initialize` function of a component (by specifying
+options in the `options` type alias of the component).  `SingletonHelloWorld`
+specifies a single option
+
+\snippet SingletonHelloWorld.hpp executable_example_options
+
+which a string specifying a name that will be placed into the constant global
+cache.  The string is fetched when performing the `PrintMessage` action. Items
+in the constant global cache are stored once per node that the executable runs
```

Comment:
"per `Charm++` node" would be more accurate, but I'm not sure this level of detail is necessary here

Reply:
true, but I don't want to get into that here